File: Difficulty: Easy/Palindrome String/palindrome-string.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String S = read.readLine();
            
            Solution ob = new Solution();
            System.out.println(ob.isPalindrome(S));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
    int isPalindrome(String S) {
        // code here
        int i = 0;
        int j = S.length()-1;
        
        String s1 = S.toLowerCase();
        
        while(i < j){
            if(s1.charAt(i) != s1.charAt(j)){
                return 0;
            }
            i++;
            j--;
        }
        return 1;
    }
};

File: Difficulty: Easy/Palindrome String/palindrome-string.py
class Solution:
    def isPalindrome(self, s):
        # code here
        n = len(s)
        
        i = 0
        j = n - 1
        
        isPalindrome = True
        
        while i < j:
            if s[i] != s[j]:
                isPalindrome = False
                break
            
            i += 1
            j -= 1
            
        return isPalindrome
        
